packages/lastrev-build-prefetch-content/src/buildTasks/writeWebsiteSectionPaths.ts

```diff
@@ -0,0 +1,124 @@
+import { getAllContentItemsForContentType } from '@last-rev/integration-contentful';
+import { each, get, set, map } from 'lodash';
+import { CONTENT_DIR, PATHS_FILE } from '../constants';
+import writeFile from '../helpers/writeFile';
+import mkdirIfNotExists from '../helpers/mkDirIfNotExists';
+import { BuildTask } from '../types';
+
+const hashResults = (results) => {
+  const out = {};
+  each(results, (result) => {
+    const id = get(result, 'sys.id');
+    if (!id) return;
+
+    out[id] = result;
+  });
+
+  return out;
+};
+
+const getDetailPagePaths = (paths, pages, websiteSections) => {
+  each(pages, (page, contentId) => {
+    const parents = get(page, 'fields.targeting');
+    const contentTypeId = get(page, 'sys.contentType.sys.id');
+    each(parents, (parent) => {
+      let parentId = get(parent, 'sys.id');
+      if (!parentId) return;
+
+      let wss;
+      let componentId;
+      let path = get(page, 'fields.slug');
+      let domain;
+      while (!domain) {
+        wss = get(websiteSections, parentId);
+        if (!wss) break;
+        if (!componentId) {
```

Comment:
Based on my comment above, I believe this could be a bug which was overlooked because of the difficulty parsing the while loop.

`componentId` can be based on a different parent than the `domain` parent?

Loops:
1) `componentId` not found, extract and save `componentId` (null result), no domain found
2) `componentId` not found, extract and save `componentId` (truthy result), no domain found
3) `componentId` found in last step, (truthy result), domain found

Now `componentId` is coming from `step2` while `domain` is coming from `step3`.

If this is the intended logic it would be great to have a comment explaining why it works that way.

Reply:
Yes, this is the intended behavior (Domain is always at the very top of the hierarchy, componentId comes from the lowest level wss of the hierarchy). I will add comments, and if possible try to refactor to a more functional approach